pkg/flist/flist.go

```diff
@@ -349,36 +350,30 @@ func random() (string, error) {
 // else it waits for the file to be deleted
 func waitPidFile(timeout time.Duration, path string, exists bool) error {
 	const delay = time.Millisecond * 100
-	cErr := make(chan error)
 
 	ctx, cancel := context.WithTimeout(context.Background(), timeout)
 	defer cancel()
 
-	go func() {
-		for {
-			select {
-			case <-ctx.Done():
-				cErr <- ctx.Err()
-			default:
-				_, err := os.Stat(path)
-				if exists {
-					if err != nil {
-						time.Sleep(delay)
-						continue
-					}
-					cErr <- nil
-				} else {
-					if err == nil {
-						time.Sleep(delay)
-						continue
-					}
-					cErr <- nil
-				}
+loop:
```

Comment:
I don't get this change. now, this function will always wait at minimum `delay`. While before it wouldn't, no ? And does the time.After in a loop is called every time or just once ?

Reply:
You are right regarding this point, I can re-fix that. But the reason I refactored the method is that It was giving a lot of weird behavior (sometimes just blocks) and it didn't require to run in a go-routine at all. This is why i added a unit-test and refactored the implementation, it's much simpler this way.

For the wait, I can change fix this to not wait on first try.